frontend/public/components/monitoring/dashboards/index.tsx

```diff
@@ -218,38 +225,59 @@ const VariableDropdown: React.FC<VariableDropdownProps> = ({ id, name, namespace
 
   const [isError, setIsError] = React.useState(false);
 
+  // const dataSourceName = variable?.datasource?.name;
+  const dataSourceName = 'test';
+  const extensions = useExtensions<DataSourceExtension>(isDataSource);
+
+  const getURL = React.useCallback(
+    async (prometheusProps) => {
+      if (!dataSourceName) {
+        return getPrometheusURL(prometheusProps);
+      } else if (!_.isEmpty(extensions)) {
+        const extension = extensions.find(
+          (ext) => ext?.properties?.contextId === 'monitoring-dashboards',
+        );
+        const getDataSource = await extension?.properties?.getDataSource();
+        const dataSource = await getDataSource(dataSourceName);
+        return getPrometheusURL(prometheusProps, dataSource?.basePath);
```

Comment:
A better way to handle errors in async functions is using a try catch block, because if there's an error the execution of this function will be interrupted and `setIsError` wont be called

Reply:
this code is now wrapped in a try/catch block